### DISCUSSION TITLE
Remove wobble

### DISCUSSION THREAD

**Discussion by IRCGraphic** [gopro/labs#171] [General]
Hello and thanks a lot for GoPro Labs, is there a way to disable internal rolling shutter correction?
I mean when I have the Max lens in circular fisheye view (without activating the Max lens mod in camera) with stabilisation turned off, only when I hit record the circular vignette starts to wobble to the camera movement, graphic example (GoPro 9): [https://youtu.be/XT-fJcrfmzU](url) (github direct links don't work)
As you may know a static object attached to the camera (being the circular rim of the Max lens) shouldn't be affected by rolling shutter as it doesn't move in relation to the sensor; I was told that this is a Gopro internal rolling shutter correction and perhaps also a gyro movement sensibility involved?
In my context, I stumbled with this issue when making a chest mounted gimbal for lowlight/night setup for cycling (stabilisation in post), as the gimbal holds very frontwards the GoPro a wider FOV is needed, as the Max lens has, to avoid a cropped framing. But there's no way to stabilise it in post because the image is wobbling all over for that supposed internal rolling shutter correction. In other words I can't shoot lowlight night video unless the internal RS correction is disabled, which would be quite nice for achieving an all terrain camera (day&night).
By disabling it also opens a broad range of functions for different uses and levels up the professional capabilities of GoPro:
· the use of Max Lens in post stabilisation or just as a traditional circular fisheye for videography
· being able to control Max lens FOV in post opens up the possibility of using extreme perspectives, with that alone gains two key benefits: a) the representation of the camera moving at speed in extreme rectilinear perspectives is much more realistic than narrow FOVs (speed look as fast as it should opposed to narrow and/or compressed perspectives) b) in a somewhat similar FOV relation the wider perspectives help stabilisation look more natural.
· technically any circular fisheye (Max lens being the case) captures more light, although projected in less pixels, so has better performance in lowlight?
· opening to the use of cinema lenses for Gopro mount, right now there's a good anamorphic lens that can't be stabilised because of this issue, people have been wanting to use Gopro for cinema for some time using cumbersome lens mods, so seems like a logical feature for a professional use of interchangeable lenses in the GoPro mount system.

Summarising I've a made downhill cycling video to illustrate some of the mentioned benefits as the speed realism and the extra FOV in a rectilinear perspective with the Max lens, it's a little bit exaggerated to show clearly the point: [https://youtu.be/wyPNHbJC-N4?t=203](url)

**Comment by dnewman-gpsw** [comment]
You're in luck, as the command already exists.  

oMBERS=1  (or !M for permanent)   - Bypass Electric Rolling Shutter correction.  

The media team as used it for product shots, never thought there would be much external interest.   I just tested it on HERO11, but it should also work in HERO10 and Mini cameras.  

So disable Hypersmooth along with this command and you should be able to do a lot of creative things.

**Reply by IRCGraphic** [reply]
No way!! :) I swear i looked all QR codes but...It's Gopro 9 compatible?

**Reply by dnewman-gpsw** [reply]
This is GP2 hack, so it would be tricky to port back to the older platform.  Also there are no Labs pending releases planned for HERO9.

**Reply by IRCGraphic** [reply]
I wish I could do it myself for my GoPro 9 If I knew how, it’s clearly a very simple non-harmful feature as disabling ERSC* releases considerable workload from the camera, I guess that’s why the GoPro 9 is already doing it in preview mode; although I was advised that it could cause bad problems if stabilisation where to be turned on at the same time; but well the only possible situation for ERSC being disabled is to have hypersmooth turned off for stabilising in computer, so the responsibility is only on the user hands.

I really wouldn’t ask if it weren’t an important feature that benefits everyone, that’s of course including Labs software (I credit GoPro Labs software disinterestedly in all my GP videos ;) and Gopro itself, it’s logic that the newer models should have better features that match their new hardware upgrades, but as you know the GoPro 9 release was the first model to have GoPro Labs implemented and the Max lens at the same time, so it’s only reasonable to ask for something that matches GoPro 9 features and hardware.

On the sensible side as a Gopro consumer I’m finding myself that I had to wait.. and learn through 2 years of what is possibly the best professional tool out there for the job, to find in the end that it’s too late/useless for the task I bought this camera for (lowlight&outdoors and stabilised superwide FOVs adding to it the game changer of speed realism**).

I forgot to mention that by disabling the ERSC there’s the additional hack of including the actual full sensor resolution? Which would be the professional thing to do for night lowlight situations, because I see this ERSC is compensating in the vertical and horizontal axis so it must be cropping the full resolution of the sensor? I’m not saying that I need the full resolution only mentioning some creative thoughts.

I’m super grateful that GoPro Labs allows for open and discrete discussions about improving software
 
*Electronic Rolling Shutter Correction
**being a cyclist and an artist myself I know there's an "uncanny valley" between engaging content and nap content and the ERSC feature allows to bridge that gap in the 1st person speed perception

**Reply by dnewman-gpsw** [reply]
I also wish Labs could see more updates to older cameras, but due to QA time and costs, Labs updates only happen when the main firmware gets updated, as Labs is typically tested in the same engineering cycles. Consider upgrading to HERO11.

**Reply by IRCGraphic** [reply]
- Well, back in the Hero 9 days Gopro Labs gave me a Lab firmware without the GoPro firmware being updated.
- Specially because of the factory malfunctions drama that was the GoPro 9 it was impossible to use the GoPro, much less having the opportunity to even request this feature back then and for the period of more over than a year.
- Personally I had to suffer 2 replacement of degenerative Gopro 9 because of factory malfunctions and finally now that I have a replacement that works, the GoPro Labs is deprecating it because it’s two generations old?
Paradoxically now the GoPro 9 is the most recommended gen because of the unreliability of the 10 and 11 to shoot for long periods of time in comparison with the "old" 9.
- So why not “consider upgrading to Hero10/11”?
 Of course I said exactly that to GoPro support team after the 3rd Gopro 9 replacement, tired of it, I offered to pay the difference for a GoPro 10 upgrade, GoPro support said no. Now that I have miraculously a camera that works, do you think anyone without an incentive would go over that absurdity again with any of the new GoPro cameras?

- Summarising why don’t GoPro Labs opens and lets install this compatible and harmless hack as the code oMBERS=1 already is being implemented and the GoPro 9 is already doing it in preview mode ?

**Reply by dnewman-gpsw** [reply]
Is not going to happen on HERO9.  I'm just being clear, as one of the main developers on this project.

**Reply by IRCGraphic** [reply]
A GoPro 11 is not going to happen either, as a matter of fact according to your negative, my only professional options are to buy an Insta 360 camera, or hack the camera by myself, the second option has more risks but how nice would it be to hack a GoPro, don’t you think?

**Reply by dnewman-gpsw** [reply]
The feature works on H10 and H11.  If you want purchase a 360 camera, ultra wide FOVs are simple with all of them.

**Reply by IRCGraphic** [reply]
No, I don’t want to buy another camera but to unlock GoPro 9 potential, as you close GoPro Labs a client flies away.
